src/gfx/renderer.cpp

```diff
@@ -16,228 +18,47 @@
 #include "texture.hpp"
 #include "texture_placeholder.hpp"
 #include "view.hpp"
-#include "view_texture.hpp"
+#include "render_target.hpp"
 #include "log.hpp"
+#include "geom.hpp"
 #include <algorithm>
 #include <magic_enum.hpp>
 #include <spdlog/spdlog.h>
 
 #define GFX_RENDERER_MAX_BUFFERED_FRAMES (2)
 
-namespace gfx {

```

Comment:
I see a lot of rewriting which is good but we need an architecture document soon. can use gitbook

Reply:
Yeah I can try something